### PR TITLE
feat: add new default tcp port parameter and generate openssl command

### PR DESCRIPTION
**Pull request by JedrzejJanasiak**
feat: add new default tcp port parameter and generate openssl command for tcp proxy api connection

## Issue

https://gravitee.atlassian.net/browse/APIM-3497

## Description

Add default tcp port parameter and generate openssl command for tcp proxy api connection on keyless plan subscription screen. Configuration option for defining custom tcp port will be covered in another separate task. 

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-dempvrsoeo.chromatic.com)

### CODE REVIEW

File: gravitee-apim-portal-webui/projects/portal-webclient-sdk/src/lib/model/listenerType.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Contact: contact@graviteesource.com
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+export type ListenerType = 'HTTP' | 'TCP' | 'SUBSCRIPTION';
+
+export const ListenerType = {
```
Comment (Okhelifi) [inline gravitee-apim-portal-webui/projects/portal-webclient-sdk/src/lib/model/listenerType.ts:18]: Has it been generated using the command "npm run update:sdk" ?

shouldn't we add the enum in the components schemas of portal-openapi.yaml and reference it in the Api model directly ?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApiMapper.java [hunk 2]
```diff
@@ -71,6 +71,10 @@ public Api convert(ExecutionContext executionContext, GenericApiEntity api) {
             List<String> entrypoints = apiEntrypoints.stream().map(ApiEntrypointEntity::getTarget).collect(Collectors.toList());
             apiItem.setEntrypoints(entrypoints);
         }
+        if (apiEntrypoints != null && !apiEntrypoints.isEmpty()) {
+            String apiListenerType = apiEntrypointService.getApiEntrypointsListenerType(api);
+            apiItem.setListenerType(Api.ListenerTypeEnum.valueOf(apiListenerType));
```
Comment (Okhelifi) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApiMapper.java]: If we change the portal-openapi.yaml you'll certainly need to change  this line to use the Enum.

File: gravitee-apim-portal-webui/projects/portal-webclient-sdk/src/lib/model/listenerType.ts (hunk 1 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-portal-webui/projects/portal-webclient-sdk/src/lib/model/listenerType.ts:18]: You're right. I added it manually :) I fixed it as you suggested so now the whole file is generated with npm run update:sdk command

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApiMapper.java (hunk 2 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/ApiMapper.java]: That's true. Now I have separate class for enum created. I fixed that line as well :)

**Review by jgiovaresco — COMMENTED** [review]
The command we show seems incomplete. We should also provide a servername

```
openssl s_client -connect localhost:4082 -servername localhost
openssl s_client -connect my-tcp-api:4082 -servername my-tcp-api
```

without it the request fails and I can see the following error in gateway logs:


```
java.lang.NullPointerException: null
    at java.base/java.util.Objects.requireNonNull(Unknown Source)
    at io.gravitee.gateway.reactor.handler.DefaultTcpAcceptor.accept(DefaultTcpAcceptor.java:53)
    at io.gravitee.gateway.reactive.reactor.handler.DefaultTcpAcceptorResolver.lambda$resolve$0(DefaultTcpAcceptorResolver.java:37)
    at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
```

**Comment by JedrzejJanasiak** [comment]
@jgiovaresco I fixed the command so now it should include servername parameter with the host name

**Review by jgiovaresco — APPROVED** [review]
👏 💪

**Pull request merged**